### PR TITLE
Robust Virtual Type

### PR DESCRIPTION
**Pull request by privat**
Improve the handling of inconsistent model when virtual types are badly defined.

The changes are mainly done with 2 things:

* still register broken `type` definition in the property, so that properties have at least a definition
* use bottom type (also called the absurd type) to indicate inconsistency in the model (instead of aborting)

Reported-by: @Morriar

### CODE REVIEW

File: src/model/model.nit [hunk 1]
```diff
@@ -817,11 +821,15 @@ abstract class MType
 		end
 
 		if sup isa MNullType then
-			# `sup` accepts only null
+			# `sup` accepts only null and bottom
```
Comment (Morriar) [inline src/model/model.nit]: The `bottom` part seems to be handled by the next if

**Comment by Morriar** [comment]
+1

**Comment by R4PaSs** [comment]
:shipit:

File: src/model/model.nit (hunk 1 above)
Comment (privat) [inline src/model/model.nit]: If fact no, but the comment does not help the reader

**Pull request merged**